src/index.ts

```diff
@@ -106,19 +108,23 @@ class MagicNixCacheAction extends DetSysAction {
   }
 
   async post(): Promise<void> {
-    if (this.noopMode) {
-      actionsCore.debug(TEXT_NOOP);
-      return;
-    }
+    if (!this.strictMode && this.errorInMain) {
+      this.exitWithWarning(`skipping post phase due to error in main phase`);
+    } else {
+      if (this.noopMode) {
```

Comment:
Nit: why does this need to be in the else branch? I feel like it made more sense to check "if no-op, then return immediately" than this